### DISCUSSION TITLE
Generation of output files

### DISCUSSION THREAD

**Discussion by Neriderc** [visjs/vis-timeline#1472] [Q&A]
Do any of the Vis.js libraries allow for output of files like JPG, PNG, PDF, etc replacing that functionally in GraphViz?

I'd like to be able to move a site away from needing GraphViz installed on the server but this is the last functionality I haven't resolved.

**Comment by anno1985** [comment]
@Neriderc did you ever manage to migrate away from GraphViz?

**Reply by Neriderc** [reply]
> @Neriderc did you ever manage to migrate away from GraphViz?

I did get off using server based GraphViz, yes, using viz.js (different from this visjs repo). Unfortunately using [an old version](https://www.npmjs.com/package/viz.js/v/2.1.2?activeTab=versions) as I wasn't able to migrate to the new one (perhaps due to a lack of effort).

With the SVG output, I'm drawing it to a canvas element in the browser then downloading this data as a JPG or PNG. I also have PDF half-working, by saving the image into a PDF file using jsPDF (this is not as good as the native PDF created by GraphViz).

I actually detect if GraphViz is installed on the server, if so I use it, if not I use the browser setup.

**Reply by anno1985** [reply]
Uhm, so, did you ever use vis-timeline or were you "just" trying to replace GraphViz with an alternative in the browser?

**Reply by Neriderc** [reply]
I was just trying to get a way to show a diagram and be able to download it as a file using the browser. I didn't get an answer to this question so never progressed very far, instead working out how to use the GraphViz-compiled-to-JS project as above.

**Comment by Thomaash** [comment]
If it's rendered on a canvas (e.g. Vis Network), it can be downloaded using native browser APIs (https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toBlob). Unfortunately Vis Timeline is built using DOM elements and there is no easy way to export it into anything else.
